### PR TITLE
Setup mypy for type checking

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Configures mypy in CI.

Expose type annotations for users.

### CODE REVIEW

File: memcache/async_memcache.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ async def flush_all(self) -> None:
         await self.writer.drain()
         response = await self.reader.readline()
         if response != b"OK\r\n":
-            raise MemcacheError(response.removesuffix(b"\r\n"))
```
Comment (Dreamsorcerer) [inline memcache/async_memcache.py:39]: This method only exists in Python 3.9+, but the library claims to support 3.6+.

File: memcache/async_memcache.py [hunk 2]
```diff
@@ -117,10 +117,10 @@ def __init__(
                 [AsyncConnection(addr, load_func=load_func, dump_func=dump_func)]
             )
 
-    def _get_connection(self, key) -> AsyncConnection:
+    def _get_connection(self, key: Union[str, bytes]) -> AsyncConnection:
         if isinstance(key, bytes):
             key = key.decode("utf-8")
-        return self._connections.get_node(key)
+        return self._connections.get_node(key)  # type: ignore[no-any-return]
```
Comment (Dreamsorcerer) [inline memcache/async_memcache.py:123]: These ignores can be removed is the hashring library is updated with type annotations.

**Comment by Dreamsorcerer** [comment]
Not sure how your build process works, but the py.typed file must be included in the package. With setup.py/cfg this would be done by setting the `include_package_data` option.

**Comment by aisk** [comment]
Thanks, LGTM, merging!

**Comment by aisk** [comment]
The CI complains `Because memcache depends on mypy (^0.930) which doesn't match any versions, version solving failed.`, maybe this version of mypy was been yanked?

**Comment by Dreamsorcerer** [comment]
No: https://pypi.org/project/mypy/0.930/#history

I'm not familiar with how poetry works. Do you need to run a command to update a lock file or something? You should be able to edit the branch if needed.

**Comment by aisk** [comment]
> Not sure how your build process works, but the py.typed file must be included in the package. With setup.py/cfg this would be done by setting the `include_package_data` option.

Found that poetry will include `py.typed` file automatically.

**Pull request merged**